### PR TITLE
Migrates V2 Select to new ui/components

### PR DESCRIPTION
**Pull request by sean-brydon**
This components moves select V2 into the new `ui/components` folder and also creates documentation behind it.
CleanShot 2022-11-07 at 13 57 38

WIP flag on variants is due to react select not handling IconStart like expected. Needs more investigation outside this PR

### CODE REVIEW

File: packages/ui/components/form/select/select.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  </Example>
+  <Example title="Focus">
+    <SelectField options={options} className="sb-pseudo--focus"/>
+  </Example> */}
+</Examples>
+
+
+## Select Story
+<Canvas>
+  <Story name="Default">
```
Comment (JeroenReumkens) [inline packages/ui/components/form/select/select.stories.mdx:112]: Are we able to still use the variants table here? Makes the layout just a bit nicer and adds the dark mode demo as well 👍

**Review by JeroenReumkens — DISMISSED** [review]
Looks great! Added one small remark to the story!

**Review by PeerRich — APPROVED** [review]

File: packages/ui/components/form/select/select.stories.mdx (hunk 1 above)
Comment (PeerRich) [reply packages/ui/components/form/select/select.stories.mdx:112]: is this blocking or can this be merged @sean-brydon @JeroenReumkens

Comment (sean-brydon) [reply packages/ui/components/form/select/select.stories.mdx:112]: I think this should be good to merge - will solve merge conflicts.

**Pull request merged**